client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -54,68 +55,39 @@ class DeployedModel:
-                          category=FutureWarning)
-        if 'model_id' in kwargs:
-            warnings.warn("`model_id` will be renamed to `run_id` in an upcoming version",
-                          category=FutureWarning)
-        host = kwargs.get('host', kwargs.get('socket', _host))
-        run_id = kwargs.get('run_id', kwargs.get('model_id', _run_id))
-        if host is None:
-            raise TypeError("missing required argument: `host`")
-        if run_id is None:
-            raise TypeError("missing required argument: `run_id`")
-
-        self._session = requests.Session()
-        if not _from_url:
-            self._session.headers.update({_utils._GRPC_PREFIX+'source': "PythonClient"})
-            try:
-                self._session.headers.update({_utils._GRPC_PREFIX+'email': os.environ['VERTA_EMAIL']})
-            except KeyError:
-                six.raise_from(EnvironmentError("${} not found in environment".format('VERTA_EMAIL')), None)
-            try:
-                self._session.headers.update({
-                    _utils._GRPC_PREFIX+'developer_key': os.environ['VERTA_DEV_KEY'],
-                    _utils._GRPC_PREFIX+'developer-key': os.environ['VERTA_DEV_KEY'],  # see Client.__init__()
-                })
-            except KeyError:
-                six.raise_from(EnvironmentError("${} not found in environment".format('VERTA_DEV_KEY')), None)
-
-        back_end_url = urlparse(host)
-        self._socket = back_end_url.netloc + back_end_url.path.rstrip('/')
-        self._scheme = back_end_url.scheme or ("https" if ".verta.ai" in self._socket else "http")
-
-        self._id = run_id
-        self._status_url = "{}://{}/api/v1/deployment/status/{}".format(self._scheme, self._socket, self._id)
-
-        self._prediction_url = None
+        self.prediction_url = prediction_url
+        creds = creds or credentials.load_from_os_env()
+        self._credentials = creds
+        self._access_token = token
+        self._session = self._make_session()
+
+    def _make_session(self):
+        session = requests.Session()
+        if self.credentials:
+            creds_headers = _utils.Connection.prefixed_headers_for_credentials(self.credentials) # TODO: extract this?
+            session.headers.update(creds_headers)
+        if self.access_token:
+            session.headers.update(AccessToken(self.access_token).headers())
+        return session
 
     def __repr__(self):
-        if self._id is not None:
-            return "<{} {}>".format(self.__class__.__name__, self._id)
-        elif self._prediction_url:
-            return "<{} at {}>".format(self.__class__.__name__, self._prediction_url)
-        else:  # if someone's messing with the object's state
-            return "<{}>".format(self.__class__.__name__)
+        return "<{} at {}>".format(self.__class__.__name__, self.prediction_url)
 
     @classmethod
-    def from_url(cls, url, token):
+    def from_url(cls, url, token=None, creds=None):
```

Comment:
Seeing this now, we could probably get rid of it and just use `__init__()` everywhere 😆 but that can wait for a future cleanup